src/day9.scala

```diff
@@ -0,0 +1,112 @@
+// using scala 3.1.0
+
+package day9
+
+import scala.collection.immutable.Queue
+import scala.util.Using
+import scala.io.Source
+
+@main def part1(): Unit =
+  println(s"The solution is ${part1(readInput())}")
+
+@main def part2(): Unit =
+  println(s"The solution is ${part2(readInput())}")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day9"))(_.mkString)
+
+type Height = Int
+case class Position(x: Int, y: Int)
+
+class Heightmap(val width: Int, val height: Int, private val data: Array[Height]):
```

Comment:
Consider using an `IArray`, if possible.

Also, what's the rationale for using a flat array, as opposed to an `Array[Array[Height]]`?

Reply:
Absolutely, thanks! I always forget `IArray`.
Concerning the nested or flat data, maybe it is more understandable for readers to use the `Array[Array[_]]`.
Do you think it would be better?

Reply:
It's not necessarily better, as long as you're consistent. Currently your various methods sometimes address things by coordinates, sometimes by indices, so it's a bit hard to follow. If you always work with indices in a flat array, it's fine. If you always go through your `apply` method, which abstracts the flat array away, it's good too. If you have an `Array[Array[_]]`, it's good too. You just need to pick one and stick to it. ;)